ansible/playbooks/koala/activestorage-migration.yml

```diff
@@ -0,0 +1,95 @@
+---
+- hosts: "all"
+  remote_user: "ansible"
+  force_handlers: true
+  become: true
+  become_user: "root"
+
+  vars_prompt:
+    - name: "confirmation"
+      default: "n"
+      prompt: "Are you sure? 'Yes' to continue."
+      private: false
+
+  tasks:
+    - assert:
+        that:
+          - "confirmation == 'Yes'"
+
+    - name: "install new dependency"
+      apt:
+        pkg: "mupdf-tools"
+        state: "present"
+
+    - name: "enable koala's maintenance mode"
+      include_tasks: ".maintenance-on.yml"
+
+    - name: "make pre-upgrade database backup to S3"
+      systemd:
+        name: "backup-databases.service"
+        state: "started"
+
+    - name: "make pre-upgrade file backup to S3"
+      systemd:
+        name: "backup-websites.service"
+        state: "started"
+
+    - name: "update Koala code"
+      git:
+        repo: "git@github.com:svsticky/constipated-koala.git"
+        dest: "/var/www/koala.{{ canonical_hostname }}"
+        version: "{{ koala.git_ref }}"
+      diff: false
+
+    - block:
+      - name: "ensure correct ruby version is installed"
+        environment:
+          PATH: "/home/koala/.rbenv/bin:{{ ansible_env.PATH }}"
+        shell: 'eval "$(rbenv init - )"; rbenv install --skip-existing'
+        args:
+          chdir: "/var/www/koala.{{ canonical_hostname }}"
+        register: "koala_ruby_result"
+        changed_when: "'Installing' in koala_ruby_result.stdout"
+
+      - name: "ensure correct bundler is present"
```

Comment:
Does this have to do with the upstream Bundler problems? Because I thought we fixed that already, but not entirely sure. 

And if we need it, shouldn't it be in the regular playbook?

Reply:
The version pin is needed because the newest version of `bundler` (>= 2.0.0) does not work with ruby 2.5, because Bundler was included in the standard library as of Ruby 2.6. This installation step is necessary because a new Ruby is installed, which does not share the set of installed gems of the previous version. I don't think that it's needed in the regular playbook, since Bundler doesn't update itself. However, I have no idea whether Bundler will be installed if a fresh copy of the playbook is run, will investigate later.